### PR TITLE
Added cosmos gRPC docs

### PR DESCRIPTION
**Pull request by eddycpc**
- Note that the `cosmos-grpc-docs.md ` file is a tool generated file

### CODE REVIEW

**Comment by lezzokafka** [comment]
Thanks @eddycpc! One quick suggestion before i go ahead: 
I think some of these items can be categorized by different [modules](https://docs.cosmos.network/master/modules/) on the [sidebar](https://github.com/crypto-com/chain-docs/blob/Add-cosmos-grpc-docs/docs/.vuepress/config.js#L58)?

**Review by leejw51crypto — APPROVED** [review]
lgtm

File: docs/resources/cosmos-grpc-docs.md [hunk 1]
```diff
@@ -0,0 +1,9298 @@
...
+
+### BaseVestingAccount
+BaseVestingAccount implements the VestingAccount interface. It contains all
+the necessary fields needed for any vesting account implementation.
+
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| base_account | [cosmos.auth.v1beta1.BaseAccount](#cosmos.auth.v1beta1.BaseAccount) |  |  |
+| original_vesting | [cosmos.base.v1beta1.Coin](#cosmos.base.v1beta1.Coin) | repeated |  |
```
Comment (lezzokafka) [inline docs/resources/cosmos-grpc-docs.md:3450]:
```suggestion
| original_vesting | [cosmos.base.v1beta1.Coin](#cosmos.base.v1beta1.Coin) | repeated | Total vesting coins upon initialization. |
```

File: docs/resources/cosmos-grpc-docs.md [hunk 2]
```diff
@@ -0,0 +1,9298 @@
...
+| delegated_free | [cosmos.base.v1beta1.Coin](#cosmos.base.v1beta1.Coin) | repeated |  |
```
Comment (lezzokafka) [inline docs/resources/cosmos-grpc-docs.md:3451]:
```suggestion
| delegated_free | [cosmos.base.v1beta1.Coin](#cosmos.base.v1beta1.Coin) | repeated | Delegated vested coins at time of delegation. |
```

File: docs/resources/cosmos-grpc-docs.md [hunk 3]
```diff
@@ -0,0 +1,9298 @@
...
+| delegated_vesting | [cosmos.base.v1beta1.Coin](#cosmos.base.v1beta1.Coin) | repeated |  |
+| end_time | [int64](#int64) |  |  |
```
Comment (lezzokafka) [inline docs/resources/cosmos-grpc-docs.md:3453]:
```suggestion
| end_time | [int64](#int64) |  | Vesting end time (UNIX Epoch time). |
```

File: docs/resources/cosmos-grpc-docs.md [hunk 4]
```diff
@@ -0,0 +1,9298 @@
...
+
+### ContinuousVestingAccount
+ContinuousVestingAccount implements the VestingAccount interface. It
+continuously vests by unlocking coins linearly with respect to time.
+
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| base_vesting_account | [BaseVestingAccount](#cosmos.vesting.v1beta1.BaseVestingAccount) |  |  |
+| start_time | [int64](#int64) |  |  |
```
Comment (lezzokafka) [inline docs/resources/cosmos-grpc-docs.md:3470]:
```suggestion
| start_time | [int64](#int64) |  | Vesting start time (UNIX Epoch time). |
```

File: docs/resources/cosmos-grpc-docs.md [hunk 5]
```diff
@@ -0,0 +1,9298 @@
...
+### PeriodicVestingAccount
+PeriodicVestingAccount implements the VestingAccount interface. It
+periodically vests by unlocking coins during each specified period.
+
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| base_vesting_account | [BaseVestingAccount](#cosmos.vesting.v1beta1.BaseVestingAccount) |  |  |
+| start_time | [int64](#int64) |  |  |
+| vesting_periods | [Period](#cosmos.vesting.v1beta1.Period) | repeated |  |
```
Comment (lezzokafka) [inline docs/resources/cosmos-grpc-docs.md:3521]:
```suggestion
| vesting_periods | [Period](#cosmos.vesting.v1beta1.Period) | repeated | The vesting schedule. |
```

File: docs/resources/cosmos-grpc-docs.md [hunk 6]
```diff
@@ -0,0 +1,9298 @@
...
+
+### PeriodicVestingAccount
+PeriodicVestingAccount implements the VestingAccount interface. It
+periodically vests by unlocking coins during each specified period.
+
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| base_vesting_account | [BaseVestingAccount](#cosmos.vesting.v1beta1.BaseVestingAccount) |  |  |
+| start_time | [int64](#int64) |  |  |
```
Comment (lezzokafka) [inline docs/resources/cosmos-grpc-docs.md:3520]:
```suggestion
| start_time | [int64](#int64) |  | The time when the coins start to vest. |
```

File: docs/resources/cosmos-grpc-docs.md [hunk 7]
```diff
@@ -0,0 +1,9298 @@
...
+
+### MsgBeginRedelegate
+MsgBeginRedelegate defines a SDK message for performing a redelegation
+of coins from a delegator and source validator to a destination validator.
+
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| delegator_address | [string](#string) |  |  |
+| validator_src_address | [string](#string) |  |  |
```
Comment (lezzokafka) [inline docs/resources/cosmos-grpc-docs.md:3554]:
```suggestion
| validator_src_address | [string](#string) |  | The source validator address. |
```

File: docs/resources/cosmos-grpc-docs.md [hunk 8]
```diff
@@ -0,0 +1,9298 @@
...
+| validator_dst_address | [string](#string) |  |  |
```
Comment (lezzokafka) [inline docs/resources/cosmos-grpc-docs.md:3555]:
```suggestion
| validator_dst_address | [string](#string) |  | The destination validators address. |
```

File: docs/resources/cosmos-grpc-docs.md [hunk 9]
```diff
@@ -0,0 +1,9298 @@
...
+| amount | [cosmos.base.v1beta1.Coin](#cosmos.base.v1beta1.Coin) |  |  |
```
Comment (lezzokafka) [inline docs/resources/cosmos-grpc-docs.md:3556]:
```suggestion
| amount | [cosmos.base.v1beta1.Coin](#cosmos.base.v1beta1.Coin) |  | The amount of redelegation. |
```

**Comment by eddycpc** [comment]
@lezzokafka 

The docs in the `cosmos-grpc-docs.md` file are auto-generated by https://github.com/pseudomuto/protoc-gen-doc.
I don't know if it's worth manually modify the auto-generated file since as the .proto files get updated often, we might always come back to generate again.

> I think some of these items can be categorized by different modules on the sidebar?

In this case, we might have to generate these files one by one

> Should #google.protobuf.* pointing to somewhere?

This might probably require us to include all third-party definitions, this file might extremely large I think.

**Comment by calvinaco** [comment]
> @lezzokafka
> 
> The docs in the `cosmos-grpc-docs.md` file are auto-generated by https://github.com/pseudomuto/protoc-gen-doc.
> I don't know if it's worth manually modify the auto-generated file since as the .proto files get updated often, we might always come back to generate again.
> 
> > I think some of these items can be categorized by different modules on the sidebar?
> 
> In this case, we might have to generate these files one by one
> 
> > Should #google.protobuf.* pointing to somewhere?
> 
> This might probably require us to include all third-party definitions, this file might extremely large I think.

I agree with @eddycpc , I think we can leave the doc this way.

**Review by calvinaco — CHANGES_REQUESTED** [review]
On the other hand, I think it is better to crate a simple scripts for generating the docs. That way everyone can generate it in the future.

**Comment by lezzokafka** [comment]
I see. Sure if it's just for reference 👯

**Review by lezzokafka — APPROVED** [review]

**Comment by eddycpc** [comment]
> On the other hand, I think it is better to crate a simple scripts for generating the docs. That way everyone can generate it in the future.

This is a good suggestion, I thought about this too but I realized it would be cool if that automation script can be done in a separate PR to keep this one small and focused on a single task.

That task to add an automated script to download the latest gRPC definitions and generate a new doc file can be a separate task on its own I believe

**Comment by lezzokafka** [comment]
> That task to add an automated script to download the latest gRPC definitions and generate a new doc file can be a separate task on its own I believe

Sure, feel free to start an separated issue for that.

**Comment by eddycpc** [comment]
> > On the other hand, I think it is better to crate a simple scripts for generating the docs. That way everyone can generate it in the future.
> 
> This is a good suggestion, I thought about this too but I realized it would be cool if that automation script can be done in a separate PR to keep this one small and focused on a single task.
> 
> That task to add an automated script to download the latest gRPC definitions and generate a new doc file can be a separate task on its own I believe

Related to #33

**Pull request merged**